backend/dataall/tasks/data_sharing/share_managers/s3_share_manager.py

```diff
@@ -142,7 +142,8 @@ def grant_target_role_access_policy(self):
                     f"arn:aws:s3:{self.dataset_region}:{self.dataset_account_id}:accesspoint/{self.access_point_name}",
                     f"arn:aws:s3:{self.dataset_region}:{self.dataset_account_id}:accesspoint/{self.access_point_name}/*"
                 ]
-                policy = existing_policy["Statement"][0]["Resource"].extend(target_resources)
+                existing_policy["Statement"][0]["Resource"].extend(target_resources)
+                policy = existing_policy
```

Comment:
I see the error that you are fixing here. Thank you for correcting the API call (https://docs.aws.amazon.com/IAM/latest/APIReference/API_PutRolePolicy.html#API_PutRolePolicy_Examples)